api/controllers/CampaignBotController.js

```diff
@@ -46,54 +46,53 @@ function CampaignBotController(campaignId) {
  * @param {object} res - Express response
  */
 CampaignBotController.prototype.chatbot = function(req, res) {
-  var self = this;
-
-  if (!self.bot) {
-    return self.handleError(req, res, 'self.bot undefined');
+  if (!this.bot) {
+    return this.handleError(req, res, 'self.bot undefined');
   }
-  if (!self.campaign) {
-    return self.handleError(req, res, 'self.campaign undefined');
+  if (!this.campaign) {
+    return this.handleError(req, res, 'self.campaign undefined');
   }
-  if (!self.mobileCommonsConfig) {
-    return self.handleError(req, res, 'self.mobileCommonsConfig undefined');
+  if (!this.mobileCommonsConfig) {
+    return this.handleError(req, res, 'self.mobileCommonsConfig undefined');
   }
-
   // This seems like it should move into router.js (or policies dir)
   if (!req.user_id) {
-    return self.handleError(req, res, 'req.user_id undefined');
+    return this.handleError(req, res, 'req.user_id undefined');
   }
 
-  logger.debug('%s incoming_message:%s', this.loggerPrefix(req),
-    req.incoming_message);
+  this.debug(req, `incoming_message_url:${req.incoming_message}`);
   if (req.incoming_image_url) {
-    logger.debug('%s incoming_image_url:%s', this.loggerPrefix(req),
-      req.incoming_image_url);
+    this.debug(req, `incoming_message_url:${req.incoming_image_url}`);
   }
 
-  dbUsers.findOne({ '_id': req.user_id }, function (err, userDoc) {
-
-    if (err) {
-      return self.handleError(req, res, err);
-    }
-
-    if (!userDoc) {
-      self.createUserAndPostSignup(req, res);
-      return;
-    }
-
-    self.user = userDoc;
-    logger.debug('%s loaded user:%s', self.loggerPrefix(req), self.user._id);
-
-    var signupId = self.user.campaigns[self.campaign._id];
-    if (!signupId) {
-      self.postSignup(req, res);
-      return;
-    }
-
-    self.loadSignup(req, res, signupId);
+  return this.loadUserAndSignup(req, res);
+}
 
-  });
+/** 
+ * Creates user and posts Signup to the Campaign.
+ * @param {object} req - Express request
+ * @param {object} res - Express response
+ */
+CampaignBotController.prototype.loadUserAndSignup = function(req, res) {
+  this.debug(req, 'loadUserAndSignup');
   
+  dbUsers
+    .findOne({ '_id': req.user_id })
+    .exec()
+    .then(userDoc => {
+      if (!userDoc) {
+        return this.createUserAndPostSignup(req, res);
+      }
+      this.user = userDoc;
```

Comment:
@sergii-tkachenko Would it make more sense to store the loaded User on our Express request `req` instead of a property on the `CampaignBotController`? refs http://stackoverflow.com/a/20710437/1470725